File: AlphaParking/AlphaParking.Models/Car.cs
﻿using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AlphaParking.Models
{
    public class Car
    {
        //формат номера: "O115AE 136"
        [Key]
        public string Number { get; set; }

        public string Brand { get; set; }
        public string Model { get; set; }

        //TODO: Добавить доп. параметры авто

        public int UserId { get; set; }
        public User User {get; set;}
        public List<ParkingSpaceCar> ParkingSpaceCars { get; set; } = new List<ParkingSpaceCar>();
    }
}

File: AlphaParking/AlphaParking.Web.Client/clientApp/models/Car.ts
import Model from './Model';

export default class Car extends Model {
    number: string;
    brand: string;
    model: string;
    userId: number;

    constructor(numberVal: string, brand: string, model: string, userId: number) {
        super();
        this.number = numberVal;
        this.brand = brand;
        this.model = model;
        this.userId = userId;
    }
}